keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -116,28 +114,33 @@ func (co *Common) SetOSInterface(intf types.OSTypeInstaller) {
 	co.OSTypeInstaller = intf
 }
 
-//GetOSVersion gets the OS name
-func GetOSVersion() string {
-	cmd := NewCommand("source /etc/os-release && echo $ID")
+//GetPackageManager get package manager of OS
+func GetPackageManager() string {
+	cmd := NewCommand("command -v apt || command -v yum")
 	err := cmd.Exec()
 	if err != nil {
 		fmt.Println(err)
 		return ""
 	}
-
-	return cmd.GetStdOut()
+	if strings.HasSuffix(cmd.GetStdOut(), APT) {
+		return APT
+	} else if strings.HasSuffix(cmd.GetStdOut(), YUM) {
+		return YUM
+	} else {
+		return ""
+	}
 }
 
 //GetOSInterface helps in returning OS specific object which implements OSTypeInstaller interface.
 func GetOSInterface() types.OSTypeInstaller {
-	switch GetOSVersion() {
-	case UbuntuOSType, RaspbianOSType, DebianOSType:
-		return &UbuntuOS{}
-	case CentOSType:
-		return &CentOS{}
+	switch GetPackageManager() {
+	case APT:
+		return &DebOS{}
+	case YUM:
+		return &RpmOS{}
 	default:
-		fmt.Printf("This OS version is currently un-supported by keadm, %s", GetOSVersion())
-		panic("This OS version is currently un-supported by keadm,")
+		fmt.Println("Failed to detect apt/yum command, exit")
```

Comment:
```suggestion
		fmt.Println("Failed to detect package manager command, exit")
```
for supporting more types?

Reply:
Maybe in other prs, `apt` and `yum` should cover most use cases.